startup.py

```diff
@@ -0,0 +1,238 @@
+        ],
+        "linux2": [
+            # example path: /opt/hfs14.0.444/bin/houdinifx
+            "/opt/hfs{version}/bin/{executable}",
+        ]
+    }
+
+    @property
+    def minimum_supported_version(self):
+        """The minimum supported Houdini version."""
+        return "12.0"
+
+    def scan_software(self):
+        """
+        Performs a scan for software installations.
+
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+
+        software_versions = []
+
+        for sw_version in self._find_software_versions():
+            if self.is_version_supported(sw_version):
+                self.logger.debug("Accepting %s", sw_version)
+                software_versions.append(sw_version)
+            else:
+                self.logger.debug("Rejecting %s", sw_version)
+                continue
+
+        return software_versions
+
+    def prepare_launch(self, exec_path, args, file_to_open=None):
+        """
+        Prepares the given software for launch
+
+        :param str exec_path: Path to DCC executable to launch
+
+        :param str args: Command line arguments as strings
+
+        :param str file_to_open: (optional) Full path name of a file to open on
+            launch
+
+        :returns: :class:`LaunchInformation` instance
+        """
+
+        tk_houdini_python_path = os.path.join(
+            self.disk_location,
+            "python",
+        )
+
+        sys.path.insert(0, tk_houdini_python_path)
+        from tk_houdini import bootstrap
+
+        # determine all environment variables
+        required_env = bootstrap.compute_environment()
+
+        # Add std context and site info to the env
+        std_env = self.get_standard_plugin_environment()
+        required_env.update(std_env)
+
+        return LaunchInformation(exec_path, args, required_env)
+
+    def _find_software_versions(self):
+        """
+        Scan the filesystem for all houdini executables.
+
+        :return: A list of :class:`SoftwareVersion` objects.
+        """
+
+        self.logger.debug("Scanning for Houdini executables...")
+
+        # use the bundled icon
+        icon_path = os.path.join(
+            self.disk_location,
+            "resources",
+            "houdini_logo.png"
+        )
+        self.logger.debug("Using icon path: %s" % (icon_path,))
+
+        if sys.platform not in self.EXECUTABLE_MATCH_TEMPLATES:
+            self.logger.debug("Houdini not supported on this platform.")
+            return []
+
+        # all the executable templates for the current OS
+        match_templates = self.EXECUTABLE_MATCH_TEMPLATES[sys.platform]
+
+        # all the discovered executables
+        all_sw_versions = []
+
+        for match_template in match_templates:
+
+            # build the glob pattern by formatting the template for globbing
+            glob_pattern = match_template.format(**self.COMPONENT_GLOB_LOOKUP)
```

Comment:
The paths for both the regex and glob are the identical except for the wildcard/match strings. This makes the code very similar across the engines with the only real difference for the default scenario is the list of templates per OS. I still think we could benefit long term by defining the known software paths as templates somewhere (in this code or perhaps externally) to make it really easy to adjust as software companies push out new install locations/hierarchies.